src/rebar_prv_dialyzer.erl

```diff
@@ -178,7 +181,12 @@ proj_plt_files(State) ->
     PltApps = get_config(State, plt_extra_apps, []),
     Apps = rebar_state:project_apps(State),
     DepApps = lists:flatmap(fun rebar_app_info:applications/1, Apps),
-    get_plt_files(BasePltApps ++ PltApps ++ DepApps, Apps).
+    DepApps1 =
+        case get_config(State, plt_include_all_deps, false) of
```

Comment:
> It would be nice not to use a boolean value and instead use an atom describing the strategy.

I also agree with this.

If there is no other opinion,  I think I will adopt @ferd's proposal.
(Because `top_level_deps` is more self-descriptive name and the default value is compatible to the current rebar3's behaviour)